survey-sdk/src/main/java/co/nimblehq/survey/sdk/network/MoshiBuilderProvider.kt

```diff
@@ -3,28 +3,43 @@ package co.nimblehq.survey.sdk.network
 import co.nimblehq.survey.sdk.entity.AnswerEntity
 import co.nimblehq.survey.sdk.entity.QuestionEntity
 import co.nimblehq.survey.sdk.entity.SurveyEntity
-import com.squareup.moshi.JsonAdapter
 import com.squareup.moshi.Moshi
 import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
 import moe.banana.jsonapi2.JsonApiConverterFactory
+import moe.banana.jsonapi2.Resource
 import moe.banana.jsonapi2.ResourceAdapterFactory
 import retrofit2.converter.moshi.MoshiConverterFactory
 
 object MoshiBuilderProvider {
 
-    fun provideJsonApiFactory(): JsonAdapter.Factory = ResourceAdapterFactory.builder()
-        .add(SurveyEntity::class.java)
-        .add(QuestionEntity::class.java)
-        .add(AnswerEntity::class.java)
+    private val listJsonApiClass = mutableListOf<Class<out Resource>>(
+        SurveyEntity::class.java,
+        QuestionEntity::class.java,
+        AnswerEntity::class.java
+    )
+
+    fun provideJsonApiFactory(): ResourceAdapterFactory = ResourceAdapterFactory.builder()
+        .add(*listJsonApiClass.toTypedArray())
         .build()
 
-    fun provideJsonApiMoshi(moshiBuilder: Moshi.Builder, factory: JsonAdapter.Factory): Moshi =
-        moshiBuilder.add((factory)).build()
+    fun provideMoshiJsonApi(
+        jsonApiFactory: ResourceAdapterFactory,
+        moshiBuilder: Moshi.Builder
+    ): Moshi = moshiBuilder
+        .add((jsonApiFactory)).build()
 
-    fun getJsonApiConverterFactory(moshi: Moshi): JsonApiConverterFactory =
-        JsonApiConverterFactory.create(moshi)
+    fun getJsonApiConverterFactory(moshiJsonApi: Moshi): JsonApiConverterFactory =
```

Comment:
why does not this method start with `provide` also for consistency?

Reply:
@luongvo It's not a big due, but seem make sense. Let change it.